File: javascript/Sorting/selectionSort.js

arr = [4, 5, 3, 1, 7, 9, 6, 20, 15]


function selectionSort(arr) {

    for (let i = 0; i < arr.length; i++) {
        let lowest = i
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[lowest]) {
                lowest = j
            }
        }
        let swap = arr[i]
        arr[i] = arr[lowest]
        arr[lowest] = swap
    }

    return arr
}

console.log(selectionSort(arr))

File: python/Sorting/selection-sort.py
def selectionSort(nums):

    for i in range(len(nums)):
        lowest = i
        for j in range(i+1,len(nums)):
            if nums[lowest] > nums[j]:
                lowest = j

        swap = nums[i]
        nums[i] = nums[lowest]
        nums[lowest] = swap

    return nums

nums = [5,3,2,10,99,456,32,21,1,4]
print(selectionSort(nums))